### DISCUSSION TITLE
Missing getSize API in expo-image library

### DISCUSSION THREAD

**Discussion by bglgwyng** [expo/expo#22795] [Archived: Questions] [answered]
I've noticed that the expo-image library (version 1.2.3) does not include the getSize API. I originally assumed that this library was intended to be a direct replacement for the Image component in React Native. Should I resort to using React Native's Image.getSize for now?

Furthermore, I came across an old pull request (PR) on GitHub (https://github.com/expo/expo/pull/14247) that aimed to add getSize support for Android. However, it appears that the PR was closed without being merged. I'm curious if anyone knows what happened to this functionality.

**Comment by hakanolgun** [comment, marked as answer]
You can use **useImage** hook from expo-image.

Example:

```
import {useImage} from "expo-image";

const image = useImage(imageUri);

useEffect(()=>{
  if(image){
    console.log("width", image.width);
    console.log("height", image.height);
  }
},[image])
```
